### ISSUE TITLE
confusion about label weight loss function

### ISSUE THREAD

**Issue by beckhamchen** [caiqi/sample-weighting-network#4]
Dear author,

       In retinanet_head_v2.py line 258, you calculate uncertainty prediction cls weights as 
 
       "uncertainty_prediction_cls = torch.ones_like(uncertainty_prediction_cls) * uncertainty_prediction_cls.mean()"

       It seems the weight for every sample is the same?

        I am confused by this code. Anything wrong here?

**Comment by caiqi** [comment]
We average the CLS weights for stable training, without which the training could fail occasionally.  For Faster and Mask R-CNN, the average is performed for positives and negatives separately. For RetinaNet, we average the CLS weights for both positives and negatives. The weights in different batches are different. We include the implementation details in Section 3.4 of the main paper.
